src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -332,30 +332,49 @@ namespace {
 
 void TwitchIrcServer::bulkRefreshLiveStatus()
 {
-    QStringList userIDs;
-    this->forEachChannel([&userIDs](ChannelPtr chan) {
+    auto twitchChans = std::make_shared<QHash<QString, TwitchChannel *>>();
+
+    this->forEachChannel([twitchChans](ChannelPtr chan) {
         auto twitchChan = dynamic_cast<TwitchChannel *>(chan.get());
-        if (!twitchChan->roomId().isEmpty())
-            userIDs.push_back(twitchChan->roomId());
+        if (twitchChan && !twitchChan->roomId().isEmpty())
+        {
+            twitchChans->insert(twitchChan->roomId(), twitchChan);
+        }
     });
 
-    for (const auto &batch : getChannelsInBatches(userIDs))
+    for (const auto &batch : getChannelsInBatches(twitchChans->keys()))
     {
         getHelix()->fetchStreams(
-            batch, QStringList(),
-            [this](std::vector<HelixStream> streams) {
+            batch, {},
+            [this, twitchChans](std::vector<HelixStream> streams) {
                 for (const auto &stream : streams)
                 {
-                    auto chan = this->getChannelOrEmpty(stream.userLogin);
-                    if (chan->getType() != Channel::Type::Twitch)
+                    if (!twitchChans->contains(stream.userId))
                         continue;
 
-                    auto twitchChan = dynamic_cast<TwitchChannel *>(chan.get());
-                    twitchChan->parseLiveStatus(true, stream);
+                    twitchChans->value(stream.userId)
+                        ->parseLiveStatus(true, stream);
+
+                    // remaining channels will be used later to set their stream status as offline
+                    twitchChans->remove(stream.userId);
                 }
             },
             []() {
                 // failure
+            },
+            [this, twitchChans] {
+                // All the channels that were not present in fetchStreams response should be assumed to be offline
+                // It is necessary to update their stream status in case they've went live -> offline
```

Comment:
```suggestion
                // It is necessary to update their stream status in case they've gone live -> offline
```
minor english xd - `they went live` is also acceptable

Reply:
Sure, titlechange_bot always posts `forsen has gone live`, not `forsen has went live` so I like this.